packages/core/src/common/test/expect.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { expect } from 'chai';
+
+export async function expectThrowsAsync(promise: Promise<any>, ...args: any[]): Promise<void> {
```

Comment:
Are you planning to adopt the API to `chai`? https://github.com/theia-ide/theia/pull/1505#discussion_r174565917

Reply:
Good point, as it crossed my mind that this import is a bit specific.

I didn't know how to allow us to use about any testing suite without getting into too much weirdness...

```ts
export async function synchronisedErrorThrower(promise: Promise<any>): Promise<Function> {
    let synchronous = () => { };
    try {
        await promise;
    } catch (e) {
        synchronous = () => { throw e; };
    } finally {
        return synchronous;
    }
}
```

Usage:
```ts
fn = await synchronizedErrorThrower(promise, Error);
expect(fn).to.throw(Error);
```

Would this be better ?

Reply:
@kittaakos 
this would be the last point to figure out I think in order to end this PR ?

Reply:
I look into your changes right now and try it out.